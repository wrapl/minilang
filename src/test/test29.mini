let T1 := table()

T1:insert("A", [1, 2, 3, 4])
T1:insert("B", ["a", "b", "c", "d"])

print(list(T1), "\n")


let T2 := table(A is [1, 2, 3, 4], B is ["a", "b", "c", "d"])

T2:put([5, "e"])
T2:put(A is 6, B is "f")
T2:push([0, "_"])
T2:insert(3, A is 99)

for Row in T2 do
	print(Row, "\n")
end


let T := table(A is list(1 .. 26), B is list(26 .. 1 by -1), C is list(1 .. 26, math::sin), D is list("A" .. "Z"))

fun by(Field) fun(X, Y) X[Field] < Y[Field]

print(T:sort(by("A")), "\n")

print(T:sort(by("B")), "\n")

print(T:sort(by("C")), "\n")

print(T:sort(by("D")), "\n")

