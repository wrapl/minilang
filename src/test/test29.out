[<A is 1, B is a>, <A is 2, B is b>, <A is 3, B is c>, <A is 4, B is d>]
<A is 0, B is _>
<A is 1, B is a>
<A is 99, B is nil>
<A is 2, B is b>
<A is 3, B is c>
<A is 4, B is d>
<A is 5, B is e>
<A is 6, B is f>
A: <1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26>
B: <26 25 24 23 22 21 20 19 18 17 16 15 14 13 12 11 10 9 8 7 6 5 4 3 2 1>
C: <0.841471 0.909297 0.14112 -0.756802 -0.958924 -0.279415 0.656987 0.989358 0.412118 -0.544021 -0.99999 -0.536573 0.420167 0.990607 0.650288 -0.287903 -0.961397 -0.750987 0.149877 0.912945 0.836656 -0.00885131 -0.84622 -0.905578 -0.132352 0.762558>
D: <A B C D E F G H I J K L M N O P Q R S T U V W X Y Z>

A: <26 25 24 23 22 21 20 19 18 17 16 15 14 13 12 11 10 9 8 7 6 5 4 3 2 1>
B: <1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26>
C: <0.762558 -0.132352 -0.905578 -0.84622 -0.00885131 0.836656 0.912945 0.149877 -0.750987 -0.961397 -0.287903 0.650288 0.990607 0.420167 -0.536573 -0.99999 -0.544021 0.412118 0.989358 0.656987 -0.279415 -0.958924 -0.756802 0.14112 0.909297 0.841471>
D: <Z Y X W V U T S R Q P O N M L K J I H G F E D C B A>

A: <11 17 5 24 23 4 18 10 12 16 6 25 22 3 19 9 13 15 7 26 21 1 2 20 8 14>
B: <16 10 22 3 4 23 9 17 15 11 21 2 5 24 8 18 14 12 20 1 6 26 25 7 19 13>
C: <-0.99999 -0.961397 -0.958924 -0.905578 -0.84622 -0.756802 -0.750987 -0.544021 -0.536573 -0.287903 -0.279415 -0.132352 -0.00885131 0.14112 0.149877 0.412118 0.420167 0.650288 0.656987 0.762558 0.836656 0.841471 0.909297 0.912945 0.989358 0.990607>
D: <K Q E X W D R J L P F Y V C S I M O G Z U A B T H N>

A: <1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26>
B: <26 25 24 23 22 21 20 19 18 17 16 15 14 13 12 11 10 9 8 7 6 5 4 3 2 1>
C: <0.841471 0.909297 0.14112 -0.756802 -0.958924 -0.279415 0.656987 0.989358 0.412118 -0.544021 -0.99999 -0.536573 0.420167 0.990607 0.650288 -0.287903 -0.961397 -0.750987 0.149877 0.912945 0.836656 -0.00885131 -0.84622 -0.905578 -0.132352 0.762558>
D: <A B C D E F G H I J K L M N O P Q R S T U V W X Y Z>

[A, B, C, D]
[A, B, C, D]
